gsa/src/web/graphql/__mocks__/tags.js

```diff
@@ -28,146 +28,89 @@ import {
   BULK_TAG,
 } from '../tags';
 
+import {createGenericQueryMock} from 'web/utils/testing';
+
 export const createTagInput = {
   name: 'foo',
   resourceType: 'OPERATING_SYSTEM',
 };
 
-export const modifyTagInput = {
-  id: '12345',
+const createTagResult = {
+  createTag: {
+    id: '12345',
+    status: 200,
+  },
 };
 
-export const createCreateTagMock = () => {
-  const createTagResult = jest.fn().mockReturnValue({
-    data: {
-      createTag: {
-        id: '12345',
-        status: 200,
-      },
-    },
-  });
-
-  const queryMock = {
-    request: {
-      query: CREATE_TAG,
-      variables: {input: createTagInput},
-    },
-    newData: createTagResult,
-  };
+export const createTagQueryMock = createGenericQueryMock(
+  CREATE_TAG,
+  createTagResult,
+  {input: createTagInput},
+);
 
-  return [queryMock, createTagResult];
+export const modifyTagInput = {
+  id: '12345',
 };
 
-export const createModifyTagMock = () => {
-  const modifyTagResult = jest.fn().mockReturnValue({
-    data: {
-      modifyTag: {
-        ok: true,
-      },
-    },
-  });
-
-  const queryMock = {
-    request: {
-      query: MODIFY_TAG,
-      variables: {input: modifyTagInput},
-    },
-    newData: modifyTagResult,
-  };
-
-  return [queryMock, modifyTagResult];
+const modifyTagResult = {
+  modifyTag: {
+    ok: true,
+  },
 };
 
-export const createEnableTagMock = tag => {
-  const enableTagResult = jest.fn().mockReturnValue({
-    data: {
-      toggleTag: {
-        ok: true,
-      },
-    },
-  });
-
-  const queryMock = {
-    request: {
-      query: TOGGLE_TAG,
-      variables: {input: {id: tag.id, active: true}},
-    },
-    newData: enableTagResult,
-  };
-  return [queryMock, enableTagResult];
+export const modifyTagQueryMock = createGenericQueryMock(
```

Comment:
Please use a function to always create new mock data for each test. We need to ensure that all tests can be run independently. It may be possible that a test changes the mocked data and therefore running tests in a undefined order some tests may fail randomly.